en/20/07.md

```diff
@@ -0,0 +1,133 @@
+
+To recap, the process is as such:
+
+1. Callee contract makes a request in a transaction
+   1. Callee contract or oracle contract emits an event
+2. Chainlink node (Off-chain) is listening for the event, where the details of the request are logged in the event
+3. In a second transaction created by the Chainlink node, it returns the data on-chain by calling a function described by the callee contract
+4. In the case of the Chainlink VRF, a randomness proof is done to ensure the number is truly random
+
+Now, similar to when we make a transaction on Ethereum or any solidity compatible blockchain, as you know, we have to pay some transaction gas. To work with oracles, we have to pay a little bit of oracle gas, also known as the [LINK or Chainlink token](https://chain.link/). The LINK token is specifically designed to work with oracles and ensure the security of the Chainlink oracle networks. Whenever we make a request following the basic request model, our contracts must be funded with a set amount of LINK, as defined by the specific oracle service that we are using (each service has different oracle gas fees). 
+
+
+<img src="./img/request-model.gif" alt="Chainlink Basic Oracle Request Model" width="469">
+
+### Why didn't we do this with the data feeds? 
+
+Now, the question then becomes, "Why wasn't this done with data feeds?" or "Why didn't we pay oracle gas for the data feeds?", and these are really good questions. With data feeds, an entire group of Chainlink nodes are being requested for data instead of just 1 Chainlink node (hence, it's a decentralized oracle network), however, only 1 entity has to kick off the request to the whole network. So we were able to benefit because someone else *already created the requests to the oracles* for the different kinds of data described in the data feeds. The data feeds are sponsored by a group of projects all making use of data feeds such as Aave, Compound, Synthetix, and more, and the updates to the network are being kicked off programmatically whenever their is a slight price change in the data they are reporting. Working together as a group, we can keep transaction costs down and create this collective good the whole ecosystem can benefit from! 
+
+In summary, someone else followed a bit more advanced version of the basic request model for us with the data feeds!
+
+### Chainlink VRF Under the Hood
+
+The Chainlink VRF follows this basic request model, with one added benefit; since their is a cryptographic proof on-chain of the randomness of the number from a Chainlink VRF node, we are safe to work with a single Chainlink VRF node! As the technology gets better and better though, even more decentralized versions of the Chainlink VRF are being created, but luckily, we are good to work with this secure method of randomness for our smart contracts. 
```

Comment:
```suggestion
The Chainlink VRF follows this basic request model, with one added benefit; since there is a cryptographic proof on-chain of the randomness of the number from a Chainlink VRF node, we are safe to work with a single Chainlink VRF node! As the technology gets better and better though, even more decentralized versions of the Chainlink VRF are being created, but luckily, we are good to work with this secure method of randomness for our smart contracts. 
```